### DISCUSSION TITLE
PutCorrelation in Spring Cloud Gateway Filter

### DISCUSSION THREAD

**Discussion by darknesstm** [apache/skywalking#10698] [General]
Using `TraceContext.putCorrelation` in the `Filter` of `Spring Cloud Gateway` is currently ineffective, although I know that `ServerWebExchange` hava a snapshot of the current tracing context, there is no way to specify it. 

Can a method similar to `TraceContext.putCorrelation(Object enhancedInstance, String key, String value)` be provided?

**Comment by wu-sheng** [comment]
I don't know what do you mean by this. 
Trace Correlation relies on the trace continuous on the thread. Otherwise, you are on your own.

**Reply by darknesstm** [reply]
example source:

```java
public class XXXGlobalFilter implements GlobalFilter{

    @Override
    public Mono<Void> filter(ServerWebExchange exchange, GatewayFilterChain chain) {
        // this is ineffective
        TraceContext.putCorrelation("aaa", "222");
       // maybe need TraceContext.putCorrelation(exchange, "aaa", "222");
        return chain.filter(exchange);
    }
}
```

**Reply by wu-sheng** [reply]
This is your own codes, you need to verify whether there is a valid tracing context.

**Reply by darknesstm** [reply]
But it seems that there is currently no way to obtain tracing context from a specified enhancement instance.

**Reply by wu-sheng** [reply]
That is common too. I think this is not guaranteed. Especially about the Spring Gateway case.
